chainer/functions/array/squeeze.py

```diff
@@ -0,0 +1,80 @@
+import six
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+def argone(iterable):
+    result = []
+    for i, x in enumerate(iterable):
+        if not isinstance(x, six.integer_types):
+            raise ValueError('elements in iterable must be int')
+        if x == 1:
+            result.append(i)
+    return result
+
+
+class Squeeze(function.Function):
+
+    """Remove demensions of size one from the shape of a ndarray."""
+
+    def __init__(self, axis=None):
+        if axis is None:
+            self.axis = None
+        elif isinstance(axis, six.integer_types):
+            self.axis = (axis,)
+        elif isinstance(axis, tuple) and all(
+                isinstance(x, six.integer_types) for x in axis):
+            self.axis = axis
+        else:
+            raise TypeError('axis must be None, int or tuple of ints')
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type = in_types[0]
+
+        if self.axis is not None:
+            for x in self.axis:
+                if x >= 0:
+                    type_check.expect(x_type.ndim > x)
```

Comment:
The fix does not work as we expect, where `and` operation raises an exception to avoid confusing results.
https://travis-ci.org/pfnet/chainer/jobs/170452193
https://github.com/pfnet/chainer/blob/master/chainer/utils/type_check.py#L154

I've split it into two expressions to be checked again.

```
if x >= 0:
    type_check.expect(x < x_type.ndim)
else:
    type_check.expect(-x_type.ndim <= x)
```